### PR TITLE
add slurm distributor for cpu and gpu workers

### PR DESCRIPTION
**Pull request by ablattmann**
WIP, needs testing.

### CODE REVIEW

File: video2dataset/main.py [hunk 1]
```diff
@@ -59,10 +62,25 @@ def video2dataset(
     optical_flow_fps: int = -1,
     optical_flow_downsample_size: int = None,
     optical_flow_dtype: type = np.float16,
+    sampler = lambda x: x,
+    slurm_cpus_per_task: int=1,
```
Comment (iejMac) [inline video2dataset/main.py:66]: not sure if this is the best way to do these args

**Comment by iejMac** [comment]
Adding some notes here:

- this might be helpful: https://github.com/mlfoundations/open_clip/blob/main/src/training/distributed.py

File: video2dataset/main.py (hunk 1 above)
Comment (ablattmann) [reply video2dataset/main.py:66]: what would you propose instead?

**Comment by SCZwangxiao** [comment]
> But then we will kinda loose control over which the distribution of shards over the processes, which I think is critical. That's why I decided to set up torch indepdently on each gpu and beforehand divide the data.

Could you explain it more clearly? Why using torch distributed will cause "But then we will kinda loose control over which the distribution of shards over the processes, which I think is critical."

I am asking because I modified the `img2dataset` using torch distributed, and the resulting webdataset files seem good.

**Pull request merged**